src/AdvApi32.Desktop/AdvApi32.Helpers.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Copyright (c) to owners found in https://github.com/AArnott/pinvoke/blob/master/COPYRIGHT.md. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+
+namespace PInvoke
+{
+    using System;
+    using System.ComponentModel;
+    using System.Runtime.InteropServices;
+
+    /// <content>
+    /// Methods and nested types that are not strictly P/Invokes but provide
+    /// a slightly higher level of functionality to ease calling into native code.
+    /// </content>
+    public static partial class AdvApi32
+    {
+        public static void InstallService(string servicePath, string serviceName, string serviceDisplayName, string serviceDescription, string userName, string password)
+        {
+            using (SafeServiceHandler hSCManager = OpenSCManager(null, null, ServiceManagerAccess.SC_MANAGER_CREATE_SERVICE))
+            {
+                if (hSCManager.IsInvalid)
+                {
+                    throw new Win32Exception(Marshal.GetHRForLastWin32Error());
```

Comment:
HRs and Win32 errors aren't the same, and `Win32Exception` takes Win32 errors. So if you were to pass a value to the ctor, you should call `Marshal.GetLastWin32Error`. But I suggest you just drop the parameter and call the default constructor. It will automatically be called.